slave.go

```diff
@@ -148,6 +155,13 @@ func (s *Slave) SetAoC(aoc bool) {
 	s.db.SetAoC(aoc)
 }
 
+// SetOnImport will set the onImport callback func
+// Note: This func is NOT thread-safe. Please only call this on initialization
+// and before any processes have begun using the database
+func (s *Slave) SetOnImport(fn func()) {
+	s.onImport = fn
```

Comment:
theoretically this is a race, because NewSlave starts the loop in a goroutine.

maybe make onImport an atomic.Value?

Reply:
True, let me think about how I want to approach this